lib/matplotlib/tests/test_backend_svg.py

```diff
@@ -588,3 +588,41 @@ def test_svg_font_string(font_str, include_generic):
 
         assert font_info == f"{size}px {font_str}"
     assert text_count == len(ax.texts)
+
+
+def test_annotationbbox_gid():
+    # Test that object gid appears in the AnnotationBbox
+    # in output svg.
+    from matplotlib.offsetbox import (OffsetImage, AnnotationBbox)
+    
+    png_path = Path(__file__).parent / \
+        "baseline_images\test_backend_svg\noscale.png"
+    fig = plt.figure()
+    ax = fig.add_subplot()
+    arr_img = plt.imread(png_path)
+    xy = (0.3, 0.55)
+
+    imagebox = OffsetImage(arr_img, zoom=0.1)
+    imagebox.image.axes = ax
+
+    ab = AnnotationBbox(imagebox, xy,
+                        xybox=(120., -80.),
+                        xycoords='data',
+                        boxcoords="offset points",
+                        pad=0.5,
+                        arrowprops=dict(
+                            arrowstyle="->",
+                            connectionstyle="angle,angleA=0,angleB=90,rad=3")
+                        )
+    ax.add_artist(ab)
+    fig.canvas.draw()
+
+    for obj in fig.findobj(include_self=True):
+        if obj.get_visible():
+            ab.set_gid("test123")
+
+    fd = BytesIO()
+    fig.savefig(fd, format='svg')
+    fd.seek(0)
+    buf = fd.read().decode()
+    fd.close()
```

Comment:
You never check if the gid you set is in the saved output?

Reply:
Thanks for the feedback, Greg. With your help I was able to fix the test so it checks the gid in the output svg, and without loading an image.

The reason i added the imports in the function is because the [test that i was modifying](https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/tests/test_backend_svg.py#L240-L241) has them included that way.